src/RA_Dlg_MemBookmark.cpp

```diff
@@ -704,50 +704,74 @@ void Dlg_MemBookmark::ExportJSON()
                 {
-                    IShellItem* pItem = nullptr;
-                    SHCreateShellItem(nullptr, nullptr, pidl, &pItem);
-                    hr = pDlg->SetDefaultFolder(pItem);
-                    if (hr == S_OK)
+					CComPtr<IShellItem> pItem;
+					SHCreateShellItem(nullptr, nullptr, pidl, &pItem );
+					
+					if (hr = pDlg->SetDefaultFolder(pItem); SUCCEEDED(hr))
                     {
                         pDlg->SetDefaultExtension(L"txt");
-                        hr = pDlg->Show(nullptr);
-                        if (hr == S_OK)
+						
+						if (hr = pDlg->Show(nullptr); SUCCEEDED(hr))
                         {
 
-                            hr = pDlg->GetResult(&pItem);
-                            if (hr == S_OK)
-                            {
+                            // Exception got thrown here (didn't before), -sbs
+                            // Expression: p==0; where p is pItem.p
+                            // Solution: reset the owned pointer
+                            auto pItem2 = std::move(pItem); // let's see if that'll do it
+
+							if (hr = pDlg->GetResult(&pItem); SUCCEEDED(hr))
+                            {                              
+                                // looks strange but the pItem still had to exist
+                                // before this block, but nullified in the
+                                // selection statement.
+                                pItem2.Release(); 
                                 LPWSTR pStr = nullptr;
-                                hr = pItem->GetDisplayName(SIGDN_FILESYSPATH, &pStr);
-                                if (hr == S_OK)
+								
+								if (hr = pItem->GetDisplayName(SIGDN_FILESYSPATH, &pStr); SUCCEEDED(hr))
                                 {
                                     Document doc;
-                                    Document::AllocatorType& allocator = doc.GetAllocator();
+                                    auto& allocator = doc.GetAllocator();
                                     doc.SetObject();
 
                                     Value bookmarks(kArrayType);
-                                    for (MemBookmark* bookmark : m_vBookmarks)
+                                    // We really shouldn't use pointers if we don't have too
+                                    // This will still use the pointer however, but that's for another time
+									for (auto bookmark : m_vBookmarks )
                                     {
                                         Value item(kObjectType);
-                                        char buffer[256];
-                                        sprintf_s(buffer, Narrow(bookmark->Description()).c_str(), sizeof(buffer));
-                                        Value s(buffer, allocator);
+										std::string buffer;
+                                        buffer.reserve(256);
```

Comment:
I realize the old code allocated 256 bytes, but since you're changing this to `std::string`, suggest letting it do its own allocation, and just truncate it too long:
```
std::string buffer = Narrow(bookmark->Description());
if (buffer.length() > 256)
    buffer.resize(256);
Value s {buffer.c_str(), allocator);
```